security-baseline/rule-types/github/osps-qa-01.yaml

```
@@ -0,0 +1,40 @@
+version: v1
+release_phase: alpha
+type: rule-type
+name: osps-qa-01
+display_name: The project's source code is publicly readable
+short_failure_message: The project's source code is not publicly readable.
+severity:
+  value: info
+context:
+  provider: github
+description: |
+  Enable users to access and review the project’s source code and
+  history, promoting transparency and collaboration within the project
+  community.
+guidance: |
+  Change repository visibility via the
+  [Settings](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/setting-repository-visibility#changing-a-repositorys-visibility)
+  page on GitHub.
+def:
+  in_entity: repository
+  rule_schema: {}
+  ingest:
+    type: git
```

It's probably better to do a single REST fetch (which may already be cached) than to do a git clone.

```suggestion
  ingest:
    type: rest
    rest:
      endpoint: "/repos/{{.Entity.Owner}}/{{.Entity.Name}}"
      parse: json
```